server/app/services/geo/esri/EsriClient.java

```diff
@@ -0,0 +1,198 @@
+          }
+        });
+
+    return responsePromise;
+  }
+
+  /**
+   * Calls the external ESRI findAddressCandidates service to retrieve address correction suggestions for the given address.
+   *
+   * <p>@see <a
+   * href="https://gisdata.seattle.gov/server/sdk/rest/index.html#/Find_Address_Candidates/02ss00000015000000/">Find
+   * Address Candidates</a>
+   * 
+   * Returns an empty optional under the following conditions:
+   * <ul>
+   *    <li>ESRI_FIND_ADDRESS_CANDIDATES_URL is not configured.</li>
+   *    <li>The external ESRI service returns an error.</li>
+   *    <li>The external ESRI services returns a non 200 status.</li>
+   * </ul>
+   * 
+   * If the external ESRI serice returns an error or non 200 status, then the request is retried up to the configured ESRI_FIND_ADDRESS_CANDIDATES_TRIES. If ESRI_FIND_ADDRESS_CANDIDATES_TRIES is not configured, the tries default to 3.
+   */
+  public CompletionStage<Optional<JsonNode>> fetchAddressSuggestions(ObjectNode addressJson) {
+    if (this.ESRI_FIND_ADDRESS_CANDIDATES_URL.isEmpty()) {
+      return CompletableFuture.completedFuture(Optional.empty());
+    }
+    WSRequest request = ws.url(this.ESRI_FIND_ADDRESS_CANDIDATES_URL.get());
+    request.setContentType(ESRI_CONTENT_TYPE);
+    request.addQueryParameter("outFields", ESRI_FIND_ADDRESS_CANDIDATES_OUT_FIELDS);
+    // "f" stands for "format", options are json and pjson (PrettyJson)
+    request.addQueryParameter("f", ESRI_FIND_ADDRESS_CANDIDATES_FORMAT);
+    String address = addressJson.findPath(AddressField.STREET.getValue()).textValue();
+    System.out.println(address);
+    String address2 = addressJson.findPath(AddressField.LINE2.getValue()).textValue();
+    String city = addressJson.findPath(AddressField.CITY.getValue()).textValue();
+    String region = addressJson.findPath(AddressField.STATE.getValue()).textValue();
+    String postal = addressJson.findPath(AddressField.ZIP.getValue()).textValue();
+    if (address != null) {
+      request.addQueryParameter("address", address);
+    }
+    if (address2 != null) {
+      request.addQueryParameter("address2", address2);
+    }
+    if (city != null) {
+      request.addQueryParameter("city", city);
+    }
+    if (region != null) {
+      request.addQueryParameter("region", region);
+    }
+    if (postal != null) {
+      request.addQueryParameter("postal", postal);
+    }
+    MutableInt tries = new MutableInt(this.ESRI_FIND_ADDRESS_CANDIDATES_TRIES);
+    return tryRequest(request, tries)
+        .thenApply(
+            res -> {
+              // return empty if still failing after retries
+              if (res.getStatus() != 200) {
+                return Optional.empty();
+              }
+              return Optional.of(res.getBody(json()));
+            });
+  }
+
+  /**
+   * Calls {@link fetchAddressSuggestions} and takes the returned address suggestions to build an {@link AddressSuggestionGroup}.
```

Comment:
suggestion (blocking): from what I can tell, this method is the only method that is intended to be consumed externally and should be the only public method. Please update the javadoc so it isn't documented in terms of reference to another method in the same class.

Reply:
Updated javadoc